### ISSUE TITLE
cabal-install Missing field in record construction configProf

### ISSUE THREAD

**Issue by claudeha** [haskell/cabal#2389]
I installed ghc-7.10.1~rc2 using ghc-7.8.2 to bootstrap.  Then cabal-install didn't work due to ghc package db format changes in 7.10, so I cabal update and cabal install -w ghc-7.8.2 cabal-install.  The new cabal-install worked, so I cabal install cabal-install again, this time with ghc-7.10.1~rc2.  Now cabal install fails:

    $ cabal sandbox init
    Writing a default package environment file to
    /home/claude/test/cabal.sandbox.config
    Creating a new sandbox at /home/claude/test/.cabal-sandbox
    $ cabal install diagrams
    Resolving dependencies...
    cabal: Distribution/Client/Config.hs:(246,37)-(299,9): Missing field in record construction configProf
    
    $ cabal --version
    cabal-install version 1.22.0.0
    using version 1.22.1.0 of the Cabal library 

The failure is independent of which package is attempted to install and whether or not there is a sandbox.  The working version I installed with ghc-7.8.2 was:

    $ cabal --version
    cabal-install version 1.22.0.0
    using version 1.22.0.0 of the Cabal library

**Comment by 23Skidoo** [comment]
Maybe has something to do with @ttuegel's recent changes.

**Comment by ttuegel** [comment]
Looks like ghc-7.10.1-rc2 ships with Cabal-1.22.1.0, which is as-yet unreleased. You will need the latest cabal-install.

@tibbe I hadn't thought of this before, but will it be a problem to release a new cabal-install at the same time as Cabal-1.22.1.0?

**Comment by claudeha** [comment]
@ttuegel thanks, I now have a working cabal-install again by bootstrapping Cabal with Setup.hs and using cabal-install's bootstrap.sh

    $ cabal --version
    cabal-install version 1.23.0.0
    using version 1.23.0.0 of the Cabal library

**Comment by 23Skidoo** [comment]
If Cabal 1.22.1 is incompatible with `cabal-install >= 1.22 && < 1.22.1` we may need to restrict `cabal-install`'s upper bounds on Hackage.

**Comment by ttuegel** [comment]
@23Skidoo Hmm, I'm not sure how I feel about that. As an alternative, I can fix Cabal-1.22.1.0 so it's API compatible *and* both sets of flags are enabled. The drawback is that I would have to disable the warning message about the deprecation of the old flags. I'm OK with removing the warning, though, because we can't start the countdown to removing the deprecated options until after Cabal-1.24, assuming we ever want to remove them at all.

**Comment by 23Skidoo** [comment]
@ttuegel Sure, if there's a way to fix Cabal-1.22.1, let's do that instead.

**Comment by ttuegel** [comment]
Fixed in 46af876.

**Comment by tibbe** [comment]
@ttuegel did this also make it into master, or isn't that needed?

**Comment by ttuegel** [comment]
It's not needed. This was a compatibility patch so that `Cabal-1.22.1.0` can be used to build `cabal-install-1.22.0.0` (to avoid breaking the PVP).

**Comment by mietek** [comment]
This appears to affect more than just building _cabal-install_.

For example:
```
$ cabal --version
cabal-install version 1.22.0.0
using version 1.22.1.0 of the Cabal library 

$ cabal unpack happy
Unpacking to happy-1.19.5/

$ cd happy-1.19.5/

$ cabal sandbox init
Writing a default package environment file to
/private/tmp/happy-1.19.5/cabal.sandbox.config
Creating a new sandbox at /private/tmp/happy-1.19.5/.cabal-sandbox

$ cabal install
Resolving dependencies...
Notice: installing into a sandbox located at
/private/tmp/happy-1.19.5/.cabal-sandbox
Configuring transformers-0.4.3.0...
cabal: Distribution/Client/Config.hs:(246,37)-(299,9): Missing field in record construction configProf
```

**Comment by ttuegel** [comment]
@mietek It's a problem with any cabal-install built with Cabal-1.22.1.0. It is patched in the 1.22 Git branch, and I have contacted GHC so they know to include it in GHC 7.10. There will also be a bug fix release.
